### PR TITLE
Verify ship status

### PR DESCRIPTION
**Pull request by KevinRansom**
Now that the SDK build process automatically publishes FSharp.Core and FSharp.Compiler.Service, we need a mechanism to remind us to rev version numbers.

Here it is.

- It uses an exact match packagereference to verify that the package id we think we are building hasn't been published yet.
- It will run automatically on  internal signed builds
- not run on PR's except this one, cos I'm verifying that it works correctly in the ci environment.

**It should be noted that** during a build, we are not connected to nuget.org by design ... it is part of our mechanism to ensure that we don't have supply chain issues accessing public dlls.

The actual projects that we build are disconnected from the general build and set the packageroot and httpcache to a discardable location, with a random name.  So ... that takes care of supply chain.

To see what it does type ````build -c release -verifypackageshipstatus````
And observe the output this is the failing case:
image

### CODE REVIEW

**Comment by KevinRansom** [comment]
Hey @brettfo , I would really appreciate, you running your eyes over this.

Thanks mate

**Comment by vzarytovskii** [comment]
We should probably have a separate CI leg which is running on the latest release branch, @KevinRansom do you want me to take care of it? I have an idea how to deal with it.

File: buildtools/checkpackages/FSharp.Compiler.Service_notshipped.fsproj [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <Import Project="$(MSBuildProjectDirectory)\..\..\eng\Versions.props" />
+ 
+  <PropertyGroup>
+    <TargetFramework>net7.0</TargetFramework>
+    <DisableImplicitFSharpCoreReference>true</DisableImplicitFSharpCoreReference>
+    <CachePath>$(MSBuildProjectDirectory)\..\..\artifacts\tmp\$([System.Guid]::NewGuid())</CachePath>
```
Comment (brettfo) [inline buildtools/checkpackages/FSharp.Compiler.Service_notshipped.fsproj:8]: These few lines shouldn't need to be here; the auto-import of `Directory.Build.props` should handle it and reduce the duplication.

File: buildtools/checkpackages/FSharp.Compiler.Service_notshipped.fsproj [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <Import Project="$(MSBuildProjectDirectory)\..\..\eng\Versions.props" />
```
Comment (brettfo) [inline buildtools/checkpackages/FSharp.Compiler.Service_notshipped.fsproj:3]: Might as well bump this import up to the `Directory.Build.props` level.

File: eng/Build.ps1 [hunk 3]
```diff
@@ -605,6 +612,39 @@ try {
         throw "Error Verifying nupkgs have access to the source code"
     }
 
+    $verifypackageshipstatusFailed = $false
+    if ($verifypackageshipstatus) {
```
Comment (brettfo) [inline eng/Build.ps1:616]: Related to Vlad's comment, pull this out into a separate file, something like `eng/VerifyPackageShipStatus.ps1`, that way it'll be easier to add as a separate build step and/or run locally.

**Review by vzarytovskii — APPROVED** [review]

File: eng/Build.ps1 (hunk 3 above)
Comment (KevinRansom) [reply eng/Build.ps1:616]: There is a build.ps1 switch for running this:
https://github.com/dotnet/fsharp/pull/13793/files#diff-9841a029538b042b58ee9aabac3d4cd9c725a2495d81d188d23f6ad617f2a262R66

https://github.com/dotnet/fsharp/pull/13793/files#diff-9841a029538b042b58ee9aabac3d4cd9c725a2495d81d188d23f6ad617f2a262R121

So build -verifypackageshipstatus   will make it easy to run.

File: buildtools/checkpackages/FSharp.Compiler.Service_notshipped.fsproj (hunk 1 above)
Comment (KevinRansom) [reply buildtools/checkpackages/FSharp.Compiler.Service_notshipped.fsproj:8]: I'm not a fan of putting project stuff in the Directory.Build.props and targets, it makes project files so hard to reason about.

**Comment by KevinRansom** [comment]
> 

@vzarytovskii , we talked about this earlier, we agreed to run this when we merge into release branches.  You can please take care of the yaml update for that :-)

**Pull request merged**